### PR TITLE
DEP: remove deprecated functions

### PR DESCRIPTION
**Pull request by rgommers**
Still a few modules to go through, but already submitting this to prevent double work in case someone else has the same idea.

### CODE REVIEW

**Comment by rgommers** [comment]
Okay, should be good to go now.

File: doc/release/1.0.0-notes.rst [hunk 1]
```diff
@@ -136,6 +136,36 @@ an array with a single element.
...
+``threshold``.
+
+The following deprecated functions have been removed from `scipy.stats.mstats`:
+``betai``, ``f_value_wilks_lambda``, ``signaltonoise`` and ``threshold``.
+
+The deprecated ``a`` and ``reta`` keywords have been removed from
+`scipy.stats.shapiro`.
+
+The deprecated functions ``sparse.csgraph.cs_graph_components`` and
+``sparse.linalg.symeig``.  have been removed from `scipy.sparse`:
```
Comment (larsoner) [inline doc/release/1.0.0-notes.rst]: weird period after `symeig` and colon at the end

**Review by larsoner — APPROVED** [review]
That is a lot of red!

+1 for merge assuming CIs are happy after addressing a small comment

File: doc/release/1.0.0-notes.rst (hunk 1 above)
Comment (rgommers) [reply doc/release/1.0.0-notes.rst]: thanks, fixed

**Comment by rgommers** [comment]
In it goes. Thanks for the review Eric.

**Pull request merged**